.github/workflows/ci.yaml

```diff
@@ -379,8 +379,111 @@ jobs:
         if-no-files-found: error
       if: ${{ env.skip_artifact_upload != 'true' }}
 
+  make_ios_matrix:
+    runs-on: ubuntu-24.04
+    outputs:
+      matrix_json: ${{ steps.make_matrix.outputs.matrix_json }}
+    steps:
+    - uses: actions/checkout@v4
+    - name: make a matrix
+      id: make_matrix
+      uses: ./.github/actions/dynamatrix
+      with:
+        matrix_yaml: |
+          include:
+          # arm64 iOS device
+          - { spec: cp313-ios_arm64_iphoneos, platform: 'iphoneos' }
+          - { spec: cp314-ios_arm64_iphoneos, platform: 'iphoneos' }
+
+          # arm64 iOS simulator
+          - { spec: cp313-ios_arm64_iphonesimulator, platform: 'iphonesimulator' }
+          - { spec: cp314-ios_arm64_iphonesimulator, platform: 'iphonesimulator' }
+
+  ios:
+    needs: [python_sdist, make_ios_matrix]
+    runs-on: macos-15
+    strategy:
+      fail-fast: false
+      matrix: ${{ fromJSON(needs.make_ios_matrix.outputs.matrix_json) }}
+
+    steps:
+    - name: fetch sdist artifact
+      id: fetch_sdist
+      uses: actions/download-artifact@v4
+      with:
+        name: ${{ needs.python_sdist.outputs.sdist_artifact_name }}
+
+    - name: install python
+      uses: actions/setup-python@v5
+      with:
+        python-version: '3.13'
+
+    - name: build wheel prereqs
+      run: |
+        set -eux
+        python3 -m pip install --user --upgrade cibuildwheel>=3.1.0
+        brew uninstall --ignore-dependencies libffi 2>&1 || true
+
+    - name: download libffi for iOS
+      env:
+        CFFI_IOS_LIBFFI_VERSION: '3.4.7-2'
+      run: |
+        set -eux
+
+        # Download prebuilt libffi from beeware/cpython-apple-source-deps
+        platform="${{ matrix.platform }}"
+        version="${CFFI_IOS_LIBFFI_VERSION}"
+        url="https://github.com/beeware/cpython-apple-source-deps/releases/download/libFFI-${version}/libffi-${version}-${platform}.arm64.tar.gz"
```

Comment:
While it looks a little weird to download binaries from beeware, `cibuildwheel` internally actually does the same for the "iOS support" `python` interpreter used to run `python -m build` with.